.cypress/integration/1_event_analytics.spec.js

```diff
@@ -280,19 +229,14 @@ describe('Saves a query on explorer page', () => {
     cy.get('[data-test-subj="eventExplorer__querySaveName"]')
       .focus()
       .type(SAVE_QUERY2, { force: true });
-
-    cy.get('[data-test-subj="eventExplorer__querySaveConfirm"]').click();
-    cy.wait(delay * 2);
-    cy.get('.euiToastHeader__title').contains('successfully').should('exist');
-
+    cy.get('[data-test-subj="eventExplorer__querySaveConfirm"]').click({ force: true });
+    cy.get('.euiToastHeader__title').should('contain', 'successfully');
```

Comment:
is it really ok to get rid of the delay here? Won't the request take some time?

Reply:
Same concern in a few places

Reply:
The wait here is like a hard wait, event request has returned successfully, it would still wait there. we don't necessary need to wait here for a specific amount of time. I think wait is something we need to almost always universally avoid in our test code, we should work on gradually refactor code to use other approaches.